### PR TITLE
perf(oauth): update schema for oauth actions

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

**Review by jayzhang — DISMISSED** [review]

File: packages/fx-core/resource/yaml-schema/yaml.schema.json [hunk 1]
```diff
@@ -1658,6 +1660,125 @@
...
+          }
+        },
+        "writeToEnvironmentFile": {
+          "type": "object",
+          "additionalProperties": false,
+          "description": "Write environment variables to environment file",
+          "required": ["registrationId"],
+          "properties": {
+            "registrationId": {
+              "description": "Required. The registration id of created API key.",
```
Comment (SLdragon) [inline packages/fx-core/resource/yaml-schema/yaml.schema.json:1729]: API key -> OAuth configuration

File: packages/fx-core/resource/yaml-schema/yaml.schema.json [hunk 2]
```diff
@@ -1658,6 +1660,125 @@
...
+              "type": "string",
+              "description": "The app ID of Oauth registration."
+            },
+            "apiSpecPath": {
+              "type": "string",
+              "description": "The path of API specification file."
+            },
+            "applicableToApps": {
+              "type": "string",
+              "description": "Which app can access the API key? Values can be \"SpecificApp\" or \"AnyApp\". Default is \"AnyApp\".",
```
Comment (SLdragon) [inline packages/fx-core/resource/yaml-schema/yaml.schema.json:1695]: same here

File: packages/fx-core/resource/yaml-schema/yaml.schema.json [hunk 3]
```diff
@@ -1658,6 +1660,125 @@
...
+              "enum": ["SpecificApp", "AnyApp"]
+            },
+            "targetAudience": {
+              "type": "string",
+              "description": "Which tenant can access the API key? Values can be \"HomeTenant\" or \"AnyTenant\". Default is \"AnyTenant\".",
```
Comment (SLdragon) [inline packages/fx-core/resource/yaml-schema/yaml.schema.json:1700]: same here

File: packages/fx-core/resource/yaml-schema/yaml.schema.json (hunk 1 above)
Comment (KennethBWSong) [reply packages/fx-core/resource/yaml-schema/yaml.schema.json]: Updated, thanks!

File: packages/fx-core/resource/yaml-schema/yaml.schema.json (hunk 2 above)
Comment (KennethBWSong) [reply packages/fx-core/resource/yaml-schema/yaml.schema.json]: Updated, thanks!

**Review by SLdragon — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**